Pcie tpr 12ch soft events 360hz ch13

bhill-slac opened the pull request:
Kukhee's commits from 2022 adding 12 PCIe TPR channels, 12 triggers, soft events, and internal 360hz ch13.
Basically same as release R2.5.3-4.0

JJL772 reviewed (approved):
LGTM. Thanks again for merging these branches!

I've left a couple of comments (more like notes), but since this isn't your code I don't expect them to be fixed

marciodo commented on tprTriggerApp/src/pcieTpr.cpp:
```diff
@@ -0,0 +1,560 @@
+                        '8', '9', 'a', 'b', 'c', 'd',
+                        '\0' };
+
+typedef struct {
+    unsigned      mask:  16;  // [15..0]     channel mask
+    unsigned      tag :   4;  // [19..16]    0: event, 1: bsa ctrl, 2: bsa event
+    unsigned      dm0 :   2;  // [21..20]    bit gap
+    unsigned      mode:   1;  // [22]        1: LCLS1, 0:LCLS2
+    unsigned      dm1 :   9;  // [31..23]    bit gap
+    unsigned      size:  32;  //  
```
Uhmmm... without `__attribute__((packed))` we are depending on luck. I also prefer uint32_t to not depend on the promises of the future C++ committee. Would you mind adding this, @bhill-slac?

JJL772 replied:
iirc struct layouts are left up to the compiler, which is worse :). I've seen some small differences in semantics between GCC and MSVC before (involving bitfields of different underlying types), but of course we only care about GCC and maybe Clang. Otherwise the rules are pretty consistent.

bhill-slac commented:
I removed this unused rdtsc() function and added the packed attribute.
Looks like the pre-release branch isn't too stale.   It's missing some work that Kukhee did last Dec to add tprValidChannels and some new substitution files.   Should I update the pre-release branch and rework this pull request to be for pre-release?

bhill-slac commented:
I was reviewing more commits from the R2.5.3-br branch for new pull requests and found the above 2 commits that fix minor issues in this pull request, so I added them here.

bhill-slac commented:
Reworked as a merge to pre-release branch.

JJL772 commented:
Merge when you're ready, we don't really have a procedure for this yet. For large PRs like this, I try to avoid merging in case the author wants to submit some last minute changes.

marciodo commented:
We usually wait a few days so everyone has a chance to review. In this case, I'd love to hear from @kukhee, but after 1 week, I think you are ok to merge. As Ernest likes to say, "silence is golden".  ;-)

We stage things in pre-release until we decide that it's a good time to merge everything with `main` and create a new release. Sometimes we need to create a release immediately after merging the feature branch with pre-release. It's a case-by-case scenario.

Bruce, thank you so much for taking the time to clean up the branches. This repository is really confusing right now.
